apps/github-stack/src/github-ineraction/service/sending-email.service.ts

```diff
@@ -1,102 +1,53 @@
-import { ConfigService } from '@nestjs/config'
-import { InjectRepository } from '@nestjs/typeorm'
-import { Repository } from 'typeorm'
-import { User } from '../../users/domain/entity/user.entity'
+import { HttpService } from "@nestjs/axios";
+import { Injectable } from "@nestjs/common";
+import { EmailData } from "../domain/interface/email.interface";
+import { firstValueFrom, lastValueFrom } from "rxjs";
+import { GitRepository } from "../domain/entity/repository.entity";
+import { ConfigService } from "@nestjs/config";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { User } from "../../users/domain/entity/user.entity";
 
 @Injectable()
 export class SendingEmailService {
-  private readonly githubApiUrl = 'https://api.github.com'
+  private readonly githubApiUrl = "https://api.github.com";
   constructor(
     private readonly httpService: HttpService,
-    private readonly configService: ConfigService,
     @InjectRepository(User)
     private readonly userRep: Repository<User>,
-    @InjectRepository(GitRepository)
-    private readonly gitRepository: Repository<GitRepository>,
   ) {}
 
   async sendingEmail(data: EmailData): Promise<string> {
-    const url = 'http://localhost:3001/sendingTestingEmail/messageRequest'
-    const response = this.httpService.post(url, data)
-    return lastValueFrom(response).then((res) => res.data)
-  }
-
-  async getLatestReliase(gitRepository: GitRepository) {
-    const token = this.configService.get<string>('GITHUB_TOKEN')
-
-    const headers = {
-      Authorization: `token ${token}`,
-    }
-    try {
-      const response = await firstValueFrom(
-        this.httpService.get(
-          `${this.githubApiUrl}/repos/${gitRepository.full_name}/releases/latest`,
-          {
-            headers,
-          },
-        ),
-      )
-
-      return response.data.name
-    } catch (error) {
-      console.log(
-        `For repository ${gitRepository.full_name} latest reliase is not found`,
-      )
-    }
-  }
-
-  async checkForUpdates() {
-    const repositories = await this.gitRepository.find({ relations: ['user'] })
-    for (const repo of repositories) {
-      const release = await this.getLatestReliase(repo)
-      if (repo.latestRelease != release) {
-        repo.latestRelease = release
-        this.gitRepository.save(repo)
-        const subject = 'Here is update from your list!'
-        const text = `Hello, it is update ${repo.name} from your Watchlist!!!`
-        const letter = {
-          from: 'aleksandr.zolotarev@abstract.rs',
-          to: repo.user.email,
-          subject: subject,
-          text: text,
-        }
-
-        await this.sendingEmail(letter)
-      }
-    }
+    const url = "http://localhost:3001/sendingTestingEmail/messageRequest";
+    const response = this.httpService.post(url, data);
+    return lastValueFrom(response).then((res) => res.data);
   }
 
   async sendMonthSummary() {
```

Comment:
Just like sendNewPassword, send months summary should get all data needed to just format and send the email.
So dont call find users here and go through each user here, instead move it outside (in scheduler) and call this method with appropriate args